tests/optimizer/nonlinear/test_trust_region.py

```diff
@@ -0,0 +1,58 @@
+import pytest
+
+import theseus as th
+import torch
+
+
+# Passing different linear_solver_cls to test with both dense/sparse linearization
+# Also test sparse [cpu/gpu] because they go through different custom backends
+@pytest.mark.parametrize("dof", [1, 8])
+@pytest.mark.parametrize(
+    "linear_solver_cls",
+    [th.CholeskyDenseSolver, th.CholmodSparseSolver, th.LUCudaSparseSolver],
```

Comment:
Do we need to test `BaSpaCho`?

Reply:
No need for `Baspacho` since it uses the same sparse linearization class.